web/source/dom/preProcessor.ts

```diff
@@ -223,6 +232,13 @@ namespace com.keyman.dom {
           e.preventDefault();
           e.stopPropagation();
         }
+
+        DOMEventHandlers.states.swallowKeypress = (Levent.Lcode != 8 ? Levent.Lcode != 0 : false);
```

Comment:
This code is convoluted to read; nested conditionals like this take a lot to think through, plus the next line repeats some of this test. Can you refactor this?

A comment on the why we don't swallow keycode 8 (backspace...) might help

Reply:
... it'd help if I understood it.  All I know is that it was part of the original mnemonic code.

I mean, I can put _that_ bit in the comment, but that's not terribly helpful.  Hence why I didn't make one.